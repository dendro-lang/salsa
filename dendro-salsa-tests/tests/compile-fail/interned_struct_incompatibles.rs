#[salsa::jar(db = Db)]
struct Jar(InternedWithRetRef, InternedWithSpecify, InternedWithNoEq, InternedWithDb, InternedWithRecover, InternedWithLru);

trait Db: salsa::DbWithJar<Jar> {}


#[salsa::interned(jar = Jar, return_ref)]
struct InternedWithRetRef {
    field: u32,    
}

#[salsa::interned(jar = Jar, specify)]
struct InternedWithSpecify {
    field: u32,    
}

#[salsa::interned(jar = Jar, no_eq)]
struct InternedWithNoEq {
    field: u32,    
}

#[salsa::interned(jar = Jar, db = Db)]
struct InternedWithDb {
    field: u32,    
}

#[salsa::interned(jar = Jar, recover_fn = recover)]
struct InternedWithRecover {
    field: u32,    
}

#[salsa::interned(jar = Jar, lru =12)]
struct InternedWithLru {
    field: u32,    
}
fn main() {}