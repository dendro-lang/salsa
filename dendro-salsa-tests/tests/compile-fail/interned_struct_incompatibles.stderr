error: `return_ref` option not allowed here
 --> tests/compile-fail/interned_struct_incompatibles.rs:7:30
  |
7 | #[salsa::interned(jar = Jar, return_ref)]
  |                              ^^^^^^^^^^

error: `specify` option not allowed here
  --> tests/compile-fail/interned_struct_incompatibles.rs:12:30
   |
12 | #[salsa::interned(jar = Jar, specify)]
   |                              ^^^^^^^

error: `no_eq` option not allowed here
  --> tests/compile-fail/interned_struct_incompatibles.rs:17:30
   |
17 | #[salsa::interned(jar = Jar, no_eq)]
   |                              ^^^^^

error: `db` option not allowed here
  --> tests/compile-fail/interned_struct_incompatibles.rs:22:30
   |
22 | #[salsa::interned(jar = Jar, db = Db)]
   |                              ^^

error: unrecognized option `recover_fn`
  --> tests/compile-fail/interned_struct_incompatibles.rs:27:30
   |
27 | #[salsa::interned(jar = Jar, recover_fn = recover)]
   |                              ^^^^^^^^^^

error: `lru` option not allowed here
  --> tests/compile-fail/interned_struct_incompatibles.rs:32:30
   |
32 | #[salsa::interned(jar = Jar, lru =12)]
   |                              ^^^

error[E0412]: cannot find type `InternedWithRetRef` in this scope
 --> tests/compile-fail/interned_struct_incompatibles.rs:2:12
  |
2 | struct Jar(InternedWithRetRef, InternedWithSpecify, InternedWithNoEq, InternedWithDb, InternedWithRecover, InternedWithLru);
  |            ^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `InternedWithSpecify` in this scope
 --> tests/compile-fail/interned_struct_incompatibles.rs:2:32
  |
2 | struct Jar(InternedWithRetRef, InternedWithSpecify, InternedWithNoEq, InternedWithDb, InternedWithRecover, InternedWithLru);
  |                                ^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `InternedWithNoEq` in this scope
 --> tests/compile-fail/interned_struct_incompatibles.rs:2:53
  |
2 | struct Jar(InternedWithRetRef, InternedWithSpecify, InternedWithNoEq, InternedWithDb, InternedWithRecover, InternedWithLru);
  |                                                     ^^^^^^^^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `InternedWithDb` in this scope
 --> tests/compile-fail/interned_struct_incompatibles.rs:2:71
  |
2 | struct Jar(InternedWithRetRef, InternedWithSpecify, InternedWithNoEq, InternedWithDb, InternedWithRecover, InternedWithLru);
  |                                                                       ^^^^^^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `InternedWithRecover` in this scope
 --> tests/compile-fail/interned_struct_incompatibles.rs:2:87
  |
2 | struct Jar(InternedWithRetRef, InternedWithSpecify, InternedWithNoEq, InternedWithDb, InternedWithRecover, InternedWithLru);
  |                                                                                       ^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `InternedWithLru` in this scope
 --> tests/compile-fail/interned_struct_incompatibles.rs:2:108
  |
2 | struct Jar(InternedWithRetRef, InternedWithSpecify, InternedWithNoEq, InternedWithDb, InternedWithRecover, InternedWithLru);
  |                                                                                                            ^^^^^^^^^^^^^^^ not found in this scope
